File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        
        
        st = [[root , float('-inf')]]
        ans = 0 
        
        while st: 
            node , maxsf = st.pop() 
            if maxsf <= node.val: 
                ans +=1 
            
            if node.left:

                st.append([node.left , max(node.val ,maxsf)])
            if node.right: 

                st.append([node.right , max(node.val ,maxsf)] )
        return ans 
    
    

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func goodNodes(root *TreeNode) int {
        
    ans := 0 
    dfs(root, root.Val , &ans)
    return ans 
}

func dfs(root *TreeNode, val int , ans *int){
    if root ==nil {
        return 
    }
    if root.Val >= val {
        *ans ++ 
    }
    maxv := val 
    if val < root.Val {
        maxv = root.Val 
    }
    dfs(root.Left , maxv , ans )
    dfs(root.Right, maxv , ans )
}

